backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/service/EdcAdapterService.java

```diff
@@ -487,6 +543,379 @@ public Response getProxyPullRequest(String url, String authKey, String authCode,
+                catalogArray = objectMapper.createArrayNode().add(catalogArray);
+            }
+            JsonNode targetCatalogEntry = null;
+            for (var entry : catalogArray) {
+                var dctTypeObject = entry.get("dct:type");
+                if (dctTypeObject != null) {
+                    if (("https://w3id.org/catenax/taxonomy#DigitalTwinRegistry").equals(dctTypeObject.get("@id").asText())) {
+                        if ("3.0".equals(entry.get("https://w3id.org/catenax/ontology/common#version").asText())) {
+                            if (targetCatalogEntry == null) {
+                                targetCatalogEntry = entry;
+                            } else {
+                                log.warn("Ambiguous catalog entries found! \n" + catalogArray.toPrettyString());
+                            }
+                        }
+                    }
+                }
+            }
+            if (targetCatalogEntry == null) {
+                log.error("Could not find asset for DigitalTwinRegistry at partner " + partner.getBpnl() + "'s catalog");
+                log.warn("CATALOG CONTENT \n" + catalogArray.toPrettyString());
+                return false;
+            }
+            String assetId = targetCatalogEntry.get("@id").asText();
+            JsonNode negotiationResponse = initiateNegotiation(partner, targetCatalogEntry);
+            String negotiationId = negotiationResponse.get("@id").asText();
+            // Await confirmation of contract and contractId
+            String contractId = null;
+            for (int i = 0; i < 100; i++) {
+                Thread.sleep(100);
+                var responseObject = getNegotiationState(negotiationId);
+                if ("FINALIZED".equals(responseObject.get("edc:state").asText())) {
+                    contractId = responseObject.get("edc:contractAgreementId").asText();
+                    break;
+                }
+            }
+            if (contractId == null) {
+                var negotiationState = getNegotiationState(negotiationId);
+                log.warn("no contract id, last negotiation state: \n" + negotiationState.toPrettyString());
+                log.error("Failed to obtain " + assetId + " from " + partner.getEdcUrl());
+                return false;
+            }
+            var contractMapping = edcContractMappingService.find(partner.getBpnl());
+            contractMapping.setDtrContractId(contractId);
+            contractMapping.setDtrAssetId(assetId);
+            edcContractMappingService.update(contractMapping);
+            log.info("Got contract for DTR api with partner " + partner.getBpnl());
+            return true;
+        } catch (Exception e) {
+            log.error("Error in Negotiation for DTR of " + partner.getBpnl(), e);
+            return false;
+        }
+    }
+
+    private boolean initiateDTRTransferForLookupApi(Partner partner, String specificAssetIdType, String specificAssetIdValue) {
```

Comment:
I added the digitalTwinType/PartType key value-pair hardcoded to the query within the initiateDTRTransferForLookupApi. Since both the getPartnerAasForMaterial  and the getPartnerAasForProduct internally use the initiateDTRTransferForLookupApi, the digitalTwinType/PartType data is always checkd.